### PR TITLE
Improvements in collision-to-TVX matching

### PR DESCRIPTION
**Pull request by ekryshen**

### CODE REVIEW

**Review by ddobrigk — APPROVED** [review]

File: Common/TableProducer/eventSelection.cxx [hunk 1]
```diff
@@ -480,6 +482,40 @@ struct EventSelectionTask {
...
+    float zVtxSigma = 2.7 * pow(nContrib, -0.466) + 0.024;
+    zVtxSigma += 1.0; // additional uncertainty due to imperfectections of FT0 time calibration
+
+    auto itMin = mapGlobalBcVtxZ.lower_bound(minBC);
+    auto itMax = mapGlobalBcVtxZ.upper_bound(maxBC);
+
+    float bestChi2 = 1e+10;
+    int64_t bestGlobalBC = 0;
+    for (std::map<int64_t, float>::iterator it = itMin; it != itMax; ++it) {
+      float chi2 = pow((it->second - zVtxCol) / zVtxSigma, 2) + pow((it->first - meanBC) / sigmaBC, 2.);
```
Comment (vkucera) [inline Common/TableProducer/eventSelection.cxx:509]: Unprotected division by 0 introduced due to casting a floating point number `weightedSigma / bcNS` < 1 to `int64_t`.

File: Common/TableProducer/eventSelection.cxx [hunk 2]
```diff
@@ -517,7 +552,7 @@ struct EventSelectionTask {
     evsel.reserve(collisions.size());
   }
 
-  void processRun2(aod::Collision const& col, BCsWithBcSelsRun2 const&, aod::Tracks const&, aod::FV0Cs const&)
+  void processRun2(aod::Collision const& col, BCsWithBcSelsRun2 const&, FullTracksIU const&, aod::FV0Cs const&)
```
Comment (vkucera) [inline Common/TableProducer/eventSelection.cxx:555]: Wrong subscription to a Run 3 table.

Comment (ekryshen) [reply Common/TableProducer/eventSelection.cxx:555]: @vkucera Thanks, indeed, Jan Fiete also pinged me on this issue. I hope to fix it before tomorrow's tag.

File: Common/TableProducer/eventSelection.cxx (hunk 1 above)
Comment (vkucera) [reply Common/TableProducer/eventSelection.cxx:509]: Fixed in https://github.com/AliceO2Group/O2Physics/pull/8032

File: Common/TableProducer/eventSelection.cxx (hunk 2 above)
Comment (vkucera) [reply Common/TableProducer/eventSelection.cxx:555]: Fixed in https://github.com/AliceO2Group/O2Physics/pull/8044

**Pull request merged**